pyiron_base/archiving/import_archive.py

```diff
@@ -0,0 +1,79 @@
+import os
+import pandas
+import numpy as np
+from shutil import rmtree
+from distutils.dir_util import copy_tree
+import tarfile
+
+
+def getdir(path):
+    path_base_name = os.path.basename(path)
+    if path_base_name == "":
+        return os.path.basename(os.path.dirname(path))
+    else:
+        return path_base_name
+
+
+def update_id_lst(record_lst, job_id_lst):
+    masterid_lst = []
+    for masterid in record_lst:
+        if np.isnan(masterid):
+            masterid_lst.append(masterid)
+        elif isinstance(masterid, int) or isinstance(masterid, float):
+            masterid = int(masterid)
+            masterid_lst.append(job_id_lst[masterid])
+    return masterid_lst
+
+
+def extract_archive(archive_directory):
+    fname = archive_directory+".tar.gz"
+    tar = tarfile.open(fname, "r:gz")
+    tar.extractall()
+    tar.close()
+
+
+def import_jobs(project_instance, directory_to_import_to, archive_directory, df, compressed=True):
+    # Copy HDF5 files
+    if isinstance(archive_directory,str):#if the archive_directory is a path
+        archive_directory = os.path.basename(archive_directory)
+    elif hasattr(archive_directory,'path'):#if the archive_directory is a project
```

Comment:
Please use the `static_isinstance()` function to check the type of a class - an example can be found here: https://github.com/pyiron/pyiron_base/blob/9fce7ef80c9e58a2a4e109ced403c3c371d7c292/pyiron_base/job/generic.py#L1050

Reply:
It is done!